server/components/posts/post-controllers.js

```diff
@@ -0,0 +1,42 @@
+const {replaceTagNameWithTagId} = require("../utils");
+const Post = require("./post-model");
+
+exports.getAllPosts = async (req, res) => {
+	const MAX_LIMIT = 50;
+	const limit = parseInt(req.query.num, 10) || MAX_LIMIT;
+
+	try {
+		const posts = await Post.find({}).limit(Math.min(Math.max(limit, 0), MAX_LIMIT));
+
+		if (posts.length > 0) {
+			res.status(200).json({posts});
+		} else {
+			res.status(200).json({message: "No posts were found"});
+		}
+	} catch (error) {
+		res.status(500).json({error});
+	}
+};
+
+exports.createPost = async (req, res) => {
+	try {
+		const tags = await replaceTagNameWithTagId(req.body.tags);
+
+		const newPost = await new Post({
+			title: req.body.title,
+			body: req.body.body,
+			isQuestion: req.body.isQuestion,
+			tags,
+			authorId: "5c07a5a54a9d0c0012cd8b35" // Fake mongo id, the real one must come from auth
+		}).save();
+
+		// Update the `replyId` property of the previous post with the `id` of the newly generated one.
+		if (!req.body.isQuestion) {
+			await Post.findByIdAndUpdate(req.body.superPostId, {$set: {replyId: newPost._id}});
```

Comment:
Can't this be done in the creation of `newPost`?

Reply:
As now that part of the code update the previous post^^ 
If you are replying to a post, the id of the post you are replying to is included into the request body ( hence the presence of the optional field [s]`repliedToId`[/s] `superPostId`) and the server will use that to update the previous post to add the id of the following post. 
To summarize: 
- if you want the id of the following post to be stored in the post document you cannot do it at creation time ( no one have replied yet so there's nothing to insert )
- if you want the id of the previous post than there's no need of any lookup ( it is already present in the optional field )
I guess it depends on how you want to implement the logic to sort the posts on the front-end right? Or did i missed something?

Reply:
Got it 👍🏻 I got a bit confused while going over the code (since I'm new to Mongo and `mongoose`).

> how you want to implement the logic to sort the posts on the front-end right?

I think the thread and post / replies in it should be rendered on the client side, and the thread data should be structured on the server side.

Let's add a new route `/post/thread/:id`, which returns the structured JSON array of the posts in a thread given the root post ID.

Reply:
Ok. i'ìll update it this afternoon^^

Reply:
I added the thread endpoint on the lookup route (since i thought that is a 'logical construction' which returns a collection of posts rather than something related to the post itself)^^
We can change it anytime, let me know what you think about it!